velox/exec/tests/utils/PrestoQueryRunner.cpp

```diff
@@ -0,0 +1,526 @@
+#include "velox/common/file/FileSystems.h"
+#include "velox/dwio/common/WriterFactory.h"
+#include "velox/exec/tests/utils/QueryAssertions.h"
+#include "velox/serializers/PrestoSerializer.h"
+
+#include <utility>
+
+using namespace facebook::velox;
+
+namespace facebook::velox::exec::test {
+
+namespace {
+
+template <typename T>
+T extractSingleValue(const std::vector<RowVectorPtr>& data) {
+  VELOX_CHECK_EQ(1, data.size());
+  VELOX_CHECK_EQ(1, data[0]->childrenSize());
+
+  auto simpleVector = data[0]->childAt(0)->as<SimpleVector<T>>();
+  VELOX_CHECK(!simpleVector->isNullAt(0));
+  return simpleVector->valueAt(0);
+}
+
+void writeToFile(
+    const std::string& path,
+    const std::vector<RowVectorPtr>& data,
+    memory::MemoryPool* pool) {
+  VELOX_CHECK_GT(data.size(), 0);
+
+  dwio::common::WriterOptions options;
+  options.schema = data[0]->type();
+  options.memoryPool = pool;
+
+  auto writeFile = std::make_unique<LocalWriteFile>(path, true, false);
+  auto sink =
+      std::make_unique<dwio::common::WriteFileSink>(std::move(writeFile), path);
+  auto writer = dwio::common::getWriterFactory(dwio::common::FileFormat::DWRF)
+                    ->createWriter(std::move(sink), options);
+  for (const auto& vector : data) {
+    writer->write(vector);
+  }
+  writer->close();
+}
+
+ByteInputStream toByteStream(const std::string& input) {
+  std::vector<ByteRange> ranges;
+  ranges.push_back(
+      {reinterpret_cast<uint8_t*>(const_cast<char*>(input.data())),
+       (int32_t)input.length(),
+       0});
+  return ByteInputStream(std::move(ranges));
+}
+
+RowVectorPtr deserialize(
+    const RowTypePtr& rowType,
+    const std::string& input,
+    memory::MemoryPool* pool) {
+  auto byteStream = toByteStream(input);
+  auto serde = std::make_unique<serializer::presto::PrestoVectorSerde>();
+  RowVectorPtr result;
+  serde->deserialize(&byteStream, pool, rowType, &result, nullptr);
+  return result;
+}
+
+class ServerResponse {
+ public:
+  explicit ServerResponse(const std::string& responseJson)
+      : response_(folly::parseJson(responseJson)) {}
+
+  void throwIfFailed() const {
+    if (response_.count("error") == 0) {
+      return;
+    }
+
+    const auto& error = response_["error"];
+
+    VELOX_FAIL(
+        "Presto query failed: {} {}", error["errorCode"], error["message"]);
+  }
+
+  std::string queryId() const {
+    return response_["id"].asString();
+  }
+
+  bool queryCompleted() const {
+    return response_.count("nextUri") == 0;
+  }
+
+  std::string nextUri() const {
+    return response_["nextUri"].asString();
+  }
+
+  std::vector<RowVectorPtr> queryResults(memory::MemoryPool* pool) const {
+    if (!response_.count("binaryData")) {
+      return {};
+    }
+
+    std::vector<std::string> names;
+    std::vector<TypePtr> types;
+    for (const auto& column : response_["columns"]) {
+      names.push_back(column["name"].asString());
+      // TODO：types.push_back(parseTypeSignature(column["type"].asString()));
```

Comment:
Can we maybe parse the JSON and use logic similar to Type::create to create Type from it?

```
            "typeSignature": {
                "rawType": "row",
                "typeArguments": [
                    {
                        "rawType": "array",
                        "typeArguments": [
                            {
                                "rawType": "map",
                                "typeArguments": [
                                    {
                                        "rawType": "varchar",
                                        "typeArguments": [],
```

Reply:
The type parser #7568 from @majetideepak is about to land, I will add `parseType(column["type"].asString())` here to get the Velox type once it is merged.